pkg/api/api.go

```diff
@@ -0,0 +1,213 @@
+			log.Info("Api server closed")
+			return nil
+		}
+		log.Error("Failed serving API", "error", err)
+		return fmt.Errorf("failed serving API: %w", err)
+	}
+}
+
+// Shutdown gracefully shuts down the api server
+// Returns an error if an error is present in the context
+// or if the server cannot be shut down
+func (a *api) Shutdown(ctx context.Context) error {
+	errC := ctx.Err()
+	log := logger.FromContext(ctx)
+	shutdownCtx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
+	defer cancel()
+	err := a.server.Shutdown(shutdownCtx)
+	if err != nil {
+		log.Error("Failed to shutdown api server", "error", err)
+		return fmt.Errorf("failed shutting down API: %w", errors.Join(errC, err))
+	}
+	return errC
+}
+
+type Route struct {
+	Path    string
+	Method  string
+	Handler http.HandlerFunc
+}
+
+// RegisterRoutes sets up all endpoint handlers for the given routes
+func (a *api) RegisterRoutes(ctx context.Context, routes ...Route) error {
+	a.router.Use(logger.Middleware(ctx))
+	for _, route := range routes {
+		switch route.Method {
+		case http.MethodGet:
+			a.router.Get(route.Path, route.Handler)
+		case http.MethodPost:
+			a.router.Post(route.Path, route.Handler)
+		case http.MethodPut:
+			a.router.Put(route.Path, route.Handler)
+		case http.MethodDelete:
+			a.router.Delete(route.Path, route.Handler)
+		case http.MethodPatch:
+			a.router.Patch(route.Path, route.Handler)
+		case "Handle":
+			a.router.Handle(route.Path, route.Handler)
+		case "HandleFunc":
+			a.router.HandleFunc(route.Path, route.Handler)
+		default:
+			return fmt.Errorf("unsupported method for %s: %s", route.Path, route.Method)
+		}
+	}
+
+	// Handles requests with simple http ok
+	// Required for global tarMan in checks
+	a.router.Handle("/", okHandler(ctx))
+
+	return nil
+}
+
+// okHandler returns a handler that will serve status ok
+func okHandler(ctx context.Context) http.Handler {
+	log := logger.FromContext(ctx)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte("ok"))
+		if err != nil {
+			log.Error("Could not write response", "error", err.Error())
+		}
+	})
+}
+
+var oapiBoilerplate = openapi3.T{
+	// this object should probably be user defined
+	OpenAPI: "3.0.0",
+	Info: &openapi3.Info{
+		Title:       "Sparrow Metrics API",
+		Description: "Serves metrics collected by sparrows checks",
+		Contact: &openapi3.Contact{
+			URL:   "https://caas.telekom.de",
+			Email: "caas-request@telekom.de",
+			Name:  "CaaS Team",
+		},
+	},
+	Paths:      make(openapi3.Paths),
+	Extensions: make(map[string]any),
+	Components: &openapi3.Components{
+		Schemas: make(openapi3.Schemas),
+	},
+	Servers: openapi3.Servers{},
+}
+
+// GenerateCheckSpecs generates the OpenAPI specifications for the given checks
+// Returns the complete OpenAPI specification for all checks
+func GenerateCheckSpecs(ctx context.Context, cks map[string]checks.Check) (openapi3.T, error) {
```

Comment:
I don't know about having this here (old function was named [Openapi](https://github.com/caas-team/sparrow/blob/main/pkg/sparrow/api.go#L150)).

Maybe it'd be better to move it here https://github.com/caas-team/sparrow/blob/2ff55bcd9d345829ff555731cd03ed66bdb57925/pkg/sparrow/handlers.go#L39 because it's only used there.

Reply:
I'll move it to the check controller with #98